content/articles/automatic-sms-verification-with-the-sms-retriever-api-in-android/index.md

```diff
@@ -0,0 +1,355 @@
+Let's add these to the build.gradle file for our app:
+```
+implementation 'com.google.android.gms:play-services-auth:19.2.0'
+implementation 'org.apache.commons:commons-lang3:3.11'
+implementation 'org.greenrobot:eventbus:3.2.0'
+```
+
+### Step 3: Setup the XML layout for our project
+
+We'll create an Edit Text in this section. This Edit text will display one-time code obtained from our SMS message.
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical"
+    tools:context=".MainActivity">
+    
+        <EditText
+            android:id="@+id/editText"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:textAlignment="center"
+            android:inputType="number"
+            android:layout_marginTop="80dp"
+            android:layout_marginStart="30dp"
+            android:layout_marginEnd="30dp"/>
+</LinearLayout>
+```
+
+### Sending the mobile number to your server
+
+In this step, you have to get the user's phone number from the `EditText`. Send this code to your verification server which will send you the one-time code. Because I don't have a verification server, I'm not going to use that method in this article. I'm going to send the SMS from another phone. The SMS I'll send will include a four-digit code. This code will be extracted and displayed on the Edit text we made in activity main.xml. We will tackle this later in the article.
+
+Once you obtain the user's mobile number from the Edit text, send it to the server perfoming the verification. The server will send back a code that to the number that was sent to it. This now will have verified the phone number to your application. To perform SMS verification on a server, check out [SMS Verification on Server]. (https://developers.google.com/identity/sms-retriever/verify)
+
+### Step 4: Getting an instance of the SmsRetriverClient object
+
+We'll first create an instance of the SmsRetrieverClient object. This is followed by invoking its initSmsRetriever instance function and adding `onSuccessListener` and `onFailureListener` to the task. We wrap all these code in a function for later use.
+
+```kotlin
+private fun initSmsListener() {
+    smsClient.startSmsRetriever()
+        .addOnSuccessListener {
+            //You can perform your tasks here
+        }.addOnFailureListener { failure ->
+            failure.printStackTrace()
+            Toast.makeText(this, failure.message, Toast.LENGTH_SHORT).show()
+        }
+}
+
+```
+
+The above function is invoked on the `onCreate()` method.
+
+```
+class MainActivity : AppCompatActivity() {
+    private lateinit var  binding: ActivityMainBinding
```

Comment:
You seem to have omitted this and several others. I'll fix them for you but next time be sure to make the requested changes where applicable.

Reply:
Okay thanks @Ericgacoki . Incase of any other changes i will rectify